_episodes/04-reproducible-envs.md

```diff
@@ -244,34 +254,66 @@ $ conda activate word-count
 ### Using conda to share a package
 
 Conda packages can be built from a *recipe* and shared on 
-[anaconda.org](https://anaconda.org/) via e.g. conda-forge, 
-[or your own private or public channel](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/create-custom-channels.html).
+[anaconda.org](https://anaconda.org/) via
+[your own private or public channel](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/create-custom-channels.html), or
+via [conda-forge](https://conda-forge.org).
 
-- [conda-forge](https://conda-forge.org) is a GitHub organization containing repositories of conda recipes. 
+- conda-forge is a GitHub organization containing repositories of conda 
+  recipes. 
+- Has become the de facto standard channel for packages.
 - Several continuous integration providers ensure that each repository 
   ("feedstock") automatically builds its own recipe on Windows, Linux and OSX.
 
+A step-by-step guide on how to contribute packages can be found in the
+[conda-forge documentation](http://conda-forge.org/docs/maintainer/adding_pkgs.html). 
+
+To get an idea of what's needed, let's have a look at the 
+[boost feedstock](https://github.com/conda-forge/boost-feedstock/tree/2ceef9da69969ab3c0ae42817574b6c5b3219c99) (a set of C++ libraries). We see that:
+- Every commit is tested on every platform.
+- There's a list of maintainers.
+- There's a [meta.yaml file](https://github.com/conda-forge/boost-feedstock/blob/2ceef9da69969ab3c0ae42817574b6c5b3219c99/recipe/meta.yaml) under the `recipe/` directory, along with (optional) `build.sh` and `bld.bat` files for building 
+  non-python code on OSX/Linux and Windows platforms.
+
+---
 
-- conda-build, conda skeleton
-- bld.bat and/or build.sh usually not needed for Python packages
+### Conda vs pip vs virtualenv vs pipenv vs poetry...
 
+- Conda arose from the Python (PyData) community, but is designed to 
+  manage packages and dependencies within any software stack (less like pip, 
+  more like a cross-platform version of apt or yum).
 
+- When it comes to packaging and dependency management in Python 
+  (which can be rather convoluted and hard to understand) many alternative 
+  tools exist.
 
+| Tool | Purpose | Comments |
+| ----- | ------ | -------- |
+| [pip](https://pypi.org/project/pip/) | Python package installer | Can be used with conda. | 
+| [virtualenv](https://virtualenv.pypa.io/en/latest/) | Tool to create isolated Python environments | Partly integrated into standard library under `venv` module. |
+| [pipenv](https://pipenv.readthedocs.io/en/latest/) | Python package and virtualenv management | Official PyPA recommendation, replaces `pip` and `virtualenv`. |
```

Comment:
I am not sure it is really a replacement. One could get the impression that `pip` and `venv` are now obsolete. I would rather say that it integrates them better and provides a higher level interface.